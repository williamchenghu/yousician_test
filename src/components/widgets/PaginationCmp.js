import React from 'react'
import PropTypes from 'prop-types'

const PaginationCmp = ({ currentPageNum, listLength, onChangeCurrentPage }) => {
  //show five items in each page
  let totalPageNumber = Math.ceil(listLength / 5)
  let renderList = []
  for (let i = 1; i <= totalPageNumber; i++) {
    renderList.push({
      pageNumber: i,
    })
  }
  return (
    <div className="pagination">
      {/* if it is not first page show next page button */}
      {currentPageNum !== 1 ? (
        <div
          id="pre"
          className="number"
          onClick={() => {
            onChangeCurrentPage(currentPageNum - 1)
          }}
        >
          &lt;
        </div>
      ) : null}

      {renderList.map((page, index) => {
        return (
          <div
            key={index}
            className={`number ${
              currentPageNum === page.pageNumber ? 'high-light' : ''
            }`}
            onClick={() => {
              onChangeCurrentPage(page.pageNumber)
            }}
          >
            {page.pageNumber}
          </div>
        )
      })}
      {/* if it is not last page show pre page button */}
      {currentPageNum !== totalPageNumber ? (
        <div
          id="next"
          className="number"
          onClick={() => {
            onChangeCurrentPage(currentPageNum + 1)
          }}
        >
          &gt;
        </div>
      ) : null}
    </div>
  )
}
PaginationCmp.propTypes = {
  onChangeCurrentPage: PropTypes.func.isRequired,
  listLength: PropTypes.number.isRequired,
  currentPageNum: PropTypes.number.isRequired,
}

export default PaginationCmp
