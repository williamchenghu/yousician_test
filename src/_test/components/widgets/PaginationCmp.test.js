import React from 'react'
import PaginationCmp from '../../../components/widgets/PaginationCmp'

describe('Cmp: PaginationCmp', () => {
  const props = {
    currentPageNum: 3,
    listLength: 7,
    onChangeCurrentPage: jest.fn(),
  }
  describe('activate: PaginationCmp', () => {
    it('should be defined', () => {
      // the component should be define
      expect(PaginationCmp).toBeDefined()
    })

    it('should render correctly', () => {
      //prepare
      let wrapper = shallow(<PaginationCmp {...props} />)
      //check
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe('actions: PaginationCmp', () => {
    it('should called onChangeCurrentPage to change the page number', () => {
      let wrapper
      wrapper = shallow(<PaginationCmp {...props} />)
      //run the function before getDefaultJobId get resovled
      wrapper.find('#pre').simulate('click')
      //check
      expect(props.onChangeCurrentPage).toHaveBeenCalledWith(2)
    })
  })
})
