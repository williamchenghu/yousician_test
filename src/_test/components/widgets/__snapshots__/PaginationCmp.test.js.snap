// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cmp: PaginationCmp activate: PaginationCmp should render correctly 1`] = `
<div
  className="pagination"
>
  <div
    className="number"
    id="pre"
    onClick={[Function]}
  >
    &lt;
  </div>
  <div
    className="number "
    key="0"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="number "
    key="1"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="number"
    id="next"
    onClick={[Function]}
  >
    &gt;
  </div>
</div>
`;
